pkg/apis/serving/k8s_validation.go

```diff
@@ -250,7 +250,8 @@ func validateEnv(ctx context.Context, envVars []corev1.EnvVar) *apis.FieldError
 
 func validateEnvFrom(envFromList []corev1.EnvFromSource) *apis.FieldError {
 	var errs *apis.FieldError
-	for i, envFrom := range envFromList {
+	for i := range envFromList {
+		envFrom := envFromList[i]
```

Comment:
What's wrong with the iterator style?

Reply:
https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable

Reply:
I was trying to understand how we had so few, but I'm guessing this is limited to things where `&` is taken or it'd be *everywhere*.  This is pleasantly small, and finally a check to keep us from making this common mistake 🤩

Reply:
Note that this matters if the function stores the object somehow, but definitely no need to have a possible bug anyway :)